### PR TITLE
Support Bitbucket repos

### PR DESCRIPTION
**Pull request by zombiezen**

### CODE REVIEW

File: handler.go [hunk 1]
```diff
@@ -38,11 +39,21 @@ func newHandler(config []byte) (*handler, error) {
...
+		case strings.HasPrefix(e.Repo, "https://github.com/"):
 			e.Display = fmt.Sprintf("%v %v/tree/master{/dir} %v/blob/master{/dir}/{file}#L{line}", e.Repo, e.Repo, e.Repo)
+		case strings.HasPrefix(e.Repo, "https://bitbucket.org"):
+			e.Display = fmt.Sprintf("%v %v/src/default{/dir} %v/src/default{/dir}/{file}#{file}-{line}", e.Repo, e.Repo, e.Repo)
+		}
+		switch {
+		case e.VCS != "":
+			// Already filled in.
+		case strings.HasPrefix(e.Repo, "https://bitbucket.org/"):
+			e.VCS = "hg"
```
Comment (rakyll) [inline handler.go]: Should we default to any VCS? Should we have different VCS defaults for different providers? GitHub is the only provider here that is exclusively git, so I understand if we default to git for them. But every other default looks unnecessary to me. Thoughts?

Comment (zombiezen) [inline handler.go]: In thinking about it, even assuming Mercurial isn't guaranteed to work out on Bitbucket &mdash; they also host Git now.  Seems like the safest behavior then is to make `newHandler` return a configuration error if we get into a case where we can't autodetect.

Does that seem too onerous for users?

**Comment by zombiezen** [comment]
Made the VCS inference fail closed instead of open.

File: handler.go [hunk 2]
```diff
@@ -37,12 +38,30 @@ func newHandler(config []byte) (*handler, error) {
...
+	for path, e := range h.m {
+		switch {
+		case e.Display != "":
+			// Already filled in.
+		case strings.HasPrefix(e.Repo, "https://github.com/"):
 			e.Display = fmt.Sprintf("%v %v/tree/master{/dir} %v/blob/master{/dir}/{file}#L{line}", e.Repo, e.Repo, e.Repo)
+		case strings.HasPrefix(e.Repo, "https://bitbucket.org"):
+			e.Display = fmt.Sprintf("%v %v/src/default{/dir} %v/src/default{/dir}/{file}#{file}-{line}", e.Repo, e.Repo, e.Repo)
+		}
+		if e.VCS == "" {
```
Comment (rakyll) [inline handler.go]: Shouldn't we remove this if clause now? It is handled by the switch below if it cannot infer it.

Comment (zombiezen) [inline handler.go]: Thanks, good catch.

**Comment by rakyll** [comment]
Nice, thanks!

**Pull request merged**